src/PockyBot.NET/Services/Pegs/PegResultsHelper.cs

```diff
@@ -0,0 +1,145 @@
+using System.Collections.Generic;
+using System.Linq;
+using PockyBot.NET.Models;
+using PockyBot.NET.Persistence.Models;
+using PockyBot.NET.Persistence.Repositories;
+
+namespace PockyBot.NET.Services.Pegs
+{
+    internal class PegResultsHelper : IPegResultsHelper
+    {
+        private readonly IConfigRepository _configRepository;
+        private readonly IPegHelper _pegHelper;
+
+        public PegResultsHelper(IConfigRepository configRepository, IPegHelper pegHelper)
+        {
+            _configRepository = configRepository;
+            _pegHelper = pegHelper;
+        }
+
+        public List<PegRecipient> MapUsersToPegRecipients(List<PockyUser> users)
+        {
+            var requireKeywords = _configRepository.GetGeneralConfig("requireValues");
+            var keywords = _configRepository.GetStringConfig("keyword").ToArray();
+            var penaltyKeywords = _configRepository.GetStringConfig("penaltyKeyword").ToArray();
+
+            return users.Select(x =>
+            {
+                var receiverLocation = x.Location?.Location;
+                var validPegs = GetValidPegs(x.PegsReceived, requireKeywords, keywords, penaltyKeywords,
+                    receiverLocation);
+                var penaltyPegs = GetPenaltyPegs(x.PegsGiven, requireKeywords, keywords, penaltyKeywords);
+                    
+                return new PegRecipient
+                {
+                    Name = x.Username,
+                    UserId = x.UserId,
+                    Location = x.Location?.Location,
+                    TotalPoints = validPegs.Sum(y => y.Weight) - penaltyPegs.Count,
+                    PegCount = validPegs.Count,
+                    PenaltyCount = penaltyPegs.Count,
+                    PegsGivenCount = x.PegsGiven.Count - penaltyPegs.Count,
+                    Pegs = validPegs,
+                    Penalties = penaltyPegs
+                };
+            }).Where(x => x.PegCount > 0 || x.PenaltyCount > 0)
+                .ToList();
+        }
+
+        public List<PegRecipient> GetWinners(List<PegRecipient> allRecipients)
+        {
+            var minimum = _configRepository.GetGeneralConfig("minimum") ?? 0;
+            var winners = _configRepository.GetGeneralConfig("winners") ?? 3;
+            var eligibleWinners = allRecipients.Where(x => x.PegsGivenCount >= minimum).ToList();
+            var topCutoff = eligibleWinners.Select(x => x.TotalPoints)
+                .OrderByDescending(x => x)
+                .Take(winners)
+                .LastOrDefault();
```

Comment:
Minor: personally prefer skip 2 get first than take 3 get last

Reply:
I very much do not. That's confusing special logic.